### ISSUE TITLE
System access control correlating schemas/tables to usernames

### ISSUE THREAD

**Issue by nabaskes** [prestodb/presto#18764]
Does anyone know if it is possible to set up System Access Control for users programmatically based on their username?  For instance, user "user_bob" can only access the schema "owner_tables_bob", and so on, for arbitrary username?

I think if I do the below, any user will have access to any "schema_" prefixed table, not just ones where the suffix matches the subset of their username.  I don't think there's any way to correlate the things matching the regexes.

```
  "schemas": [
    {
      "user": ".*"
      "schema": "schema_.*",
      "owner": true
    }
  ]
```

**Comment by pratyakshsharma** [comment]
@agrawalreetika Can you help here?

**Comment by nabaskes** [comment]
Hi @agrawalreetika, to confirm, there is no out of the box way to assign users by default to schemas via a pattern on their username?

For instance, I would be assigned to u_nabaskes and you would be assigned to u_agrawalreetika?

Fine if not, appreciate your time in answering

**Comment by pratyakshsharma** [comment]
can we close this issue now? @agrawalreetika @nabaskes

**Comment by agrawalreetika** [comment]
@nabaskes I don't think it would be possible for File Based Access control.

On the other hand, if you make use of Ranger-based authorization then it should be possible, it allows {USER} as a Variable in the policies. 
More details - https://docs.cloudera.com/HDPDocuments/HDP3/HDP-3.0.0/authorization-ranger/content/resource_policy_wildcard_and_variable_reference_information.html

In Presto Ranger Based Authorization is supported for Hive Connector, if you want to try that out - https://prestodb.io/docs/current/connector/hive-security.html#ranger-based-authorization

**Comment by nabaskes** [comment]
Yeah, thanks @agrawalreetika , ended up doing that.  Thanks!

**Comment by pratyakshsharma** [comment]
Glad that you were unblocked. Please feel free to reopen or raise a new issue in case of more queries.
